### DISCUSSION TITLE
Can I follow a guide to build the Ref Arch from scratch?

### DISCUSSION THREAD

**Discussion by zackproser** [gruntwork-io/knowledge-base#58] [Help] [answered]
A customer asked: 
> Can I read some documentation to build the Ref Arch myself if I only purchase the Gruntwork subscription?

**Comment by zackproser** [comment, marked as answer]
While we have put thought into allowing users to do this, and have experimented with its feasibility internally, this is not something we currently support. We would like to support it in the future, with at least one "flavor" of the Ref Arch. 

In the past, we've directed our limited resources toward improving the Ref Arch, offering new frequently requested features, and improving our internal tooling to increase the speed at which we're able to offer the Ref Arch. As a result, we've not yet been able to prioritize a complete end-to-end guide that has been thoroughly tested and verified.

That said, we do publish [complete production ready examples](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-production) for all of the modules in our Service Catalog, for anyone who may wish to experiment and learn more this way. 

The Ref Arch itself ships with documentation explaining how to get up and running with your deployment and accomplish common development and maintenance tasks.

**Comment by rhoboat** [comment]
Unfortunately, we don't really have a full guide written about how to deploy the Reference Architecture from scratch. That said, for the most part deployment can be done by doing it in the following order:

1. Deploy account-baseline of the logs account using terragrunt apply.
1. Deploy account-baseline of the security and shared account using terragrunt apply.
1. Optional if you want to deploy Gruntwork Pipelines:
    1. Deploy ecr-repos in the shared account
    1. Build and push the docker containers for ECS Deploy Runner: ecs-deploy-runner and kaniko
1. Run terragrunt run-all apply in the logs, security, and shared folders.
1. When you are ready to deploy the application accounts:
    1. Deploy account-baseline for the account.
    1. Create any EC2 keypairs that your architecture needs. This applies to any module that uses a non-empty string for key_pair_name.
    1. Deploy everything else using terragrunt run-all apply in the application account folder.
1. Once you are ready to migrate an application to a deployed Reference Architecture, you can follow https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/c417871de44527e3b7cc1307757804df7e7c41a1/examples/for-production/infrastructure-live/docs/03-deploy-apps.md.

credit to @yorinasub17
